cpp/daal/src/algorithms/pca/oneapi/pca_dense_correlation_distr_step2_kernel_ucapi_impl.i

```diff
@@ -0,0 +1,76 @@
+/* file: pca_dense_correlation_distr_step2_kernel_ucapi_impl.i */
+/*******************************************************************************
+* Copyright 2019-2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of PCA Distributed Kernel for GPU.
+//--
+*/
+
+#ifndef __PCA_DENSE_CORRELATION_BATCH_KERNEL_UCAPI_IMPL__
+#define __PCA_DENSE_CORRELATION_BATCH_KERNEL_UCAPI_IMPL__
+
+#include "src/externals/service_ittnotify.h"
+DAAL_ITTNOTIFY_DOMAIN(pca.dense.correlation.distr.oneapi);
+
+#include "services/env_detect.h"
+#include "include/services/internal/sycl/types.h"
+#include "src/algorithms/pca/oneapi/cl_kernels/pca_cl_kernels.cl"
+#include "data_management/data/internal/numeric_table_sycl_homogen.h"
+#include "src/sycl/blas_gpu.h"
+#include "src/sycl/reducer.h"
+#include "src/algorithms/covariance/oneapi/covariance_oneapi_impl.i"
+
+using namespace daal::services;
+using namespace daal::internal;
+using namespace daal::services::internal::sycl;
+using namespace daal::data_management;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace pca
+{
+namespace internal
+{
+template <typename algorithmFPType>
+PCACorrelationKernelDistrUCAPI<algorithmFPType>::compute(DistributedInput<correlationDense> * input, PartialResult<correlationDense> * partialResult,
```

Comment:
Is it possible to use the existing online kernel for this second step instead of creating this new one?

Reply:
resolved offline